tests/python/test_monotone_constraints.py

```diff
@@ -61,62 +60,29 @@ def is_correctly_constrained(learner):
 training_dset = xgb.DMatrix(x, label=y)
 
 
-class TestMonotoneConstraints(unittest.TestCase):
+class TestMonotoneConstraints:
 
-    def test_monotone_constraints_for_exact_tree_method(self):
-
-        # first check monotonicity for the 'exact' tree method
-        params_for_constrained_exact_method = {
-            'tree_method': 'exact', 'verbosity': 1,
-            'monotone_constraints': '(1, -1)'
+    @pytest.mark.parametrize('tree_method,extra_kwargs',
```

Comment:
Why do you need a extra kwargs?

Reply:
We can't set grow_policy when tree_method is not hist.